sound/soc/sof/pcm.c

```diff
@@ -619,6 +619,33 @@ static int sof_pcm_new(struct snd_soc_component *component,
 	return 0;
 }
 
+static int ssp_dai_config_pcm_params_match(struct snd_sof_dev *sdev, const char* name,
+					   struct snd_pcm_hw_params *params)
+{
+	struct sof_ipc_dai_config *config;
+	struct snd_sof_dai *dai;
+	int number_configs;
+	int ret = -EINVAL;
+	int i;
+
+	/* we might have several dai's associated with same link, need to loop all */
+	list_for_each_entry(dai, &sdev->dai_list, list) {
+		if (dai->name && (strcmp(name, dai->name) == 0)) {
```

Comment:
!strcmp(...)?

Reply:
ok, this was a direct copy from existing function but can change.